homeworks/HW2-final/P6.py

```diff
@@ -0,0 +1,46 @@
+import numpy as np
+import time
+
+# below the timer function is adopted from the code in lecture5
+def timer(f):
+    def inner(*args):
+        t0 = time.time()
+        output = f(*args)
+        elapsed = time.time() - t0
+        print("Time Elapsed", elapsed)
+        return output
+    return inner
+
+def my_ave(areas):
+    my_sum = 0
+    for a in areas:
+        my_sum += a
+    my_sum = my_sum / len(areas)
+    #print("Myavg :",my_sum)
+    return my_sum
+
+
+def np_ave (areas):
+    return np.mean(areas)
+
+file = open('circles.txt')
+list_str= file.read().splitlines()
+radius = [ float(x) for x in list_str ]
+
+areas = [0.0]*len(radius)
+areas_np = np.zeros(len(radius))
+for i in range(len(radius)):
```

Comment:
Use `enumerate` when you are looping over a list and want both the index and item.